### ISSUE TITLE
Build breakage on ARMv6 devices with v0.11 branch

### ISSUE THREAD

**Issue by adammw** [nodejs/node-v0.x-archive#7222]
I've just been [pinged on Twitter](https://twitter.com/sageahn/status/439056429010272256) that the [v0.11.9 build here](https://gist.github.com/adammw/3245130) throws an illegal instruction on ARMv6 devices (basically the Raspberry Pi, I haven't tested any other devices). Seems to do the same on all of the ["offical" builds](http://nodejs.org/dist/) past v0.11.8.

Running `node --v8-options` presents:

```
target arm v6 vfp2 hard
ARMv7=1 VFP3=0 VFP32DREGS=0 NEON=0 SUDIV=0 UNALIGNED_ACCESSES=1 MOVW_MOVT_IMMEDIATE_LOADS=0 USE_EABI_HARDFLOAT=1
```

On v0.11.7:

```
target arm v6 vfp2 hard
ARMv7=0 VFP3=0 VFP32DREGS=0 NEON=0 SUDIV=0 UNALIGNED_ACCESSES=0 MOVW_MOVT_IMMEDIATE_LOADS=0 USE_EABI_HARDFLOAT=1
```

Obviously, an armv6 shouldn't be setting the ARMv7 flag, and that's causing illegal instructions to be generated.

I've been able to bisect it to something being changed between v0.11.7 and v0.11.8, but I haven't yet found the commit or change that needs reverting.

**Comment by wilson0x4d** [comment]
Cross-posted from another thread:

I also had previously edited `configure` to force the use of "vfpv2" instead of "vfpv3" by changing:

`o['variables']['arm_fpu'] = 'vfpv3'  # V8 3.18 no longer supports VFP2.`
to
`o['variables']['arm_fpu'] = 'vfpv2'  # V8 3.18 no longer supports VFP2.`

In addition to the armv6 change to `cpu.cc`. 

Since "vfpv2" isn't considered supported in "v8" I have not generated a patch (I'm really not sure what the correct action is wrt the configure script tbh.) My limited understanding is that 'vfp' are floating point extensions, and my assumption here is that by specifying `vfpv2` the node build is omitting said extensions (as the `VFP3=0` from --v8-options suggests.) Thus, someone will probably want to generate an appropriate patch to `configure`, or perhaps give me some insight into what scenarios it is acceptable (or unacceptable) to opt out of VFPv3.

**Comment by kevinhikaruevans** [comment]
Was this patched in master?

**Comment by jasnell** [comment]
@adammw ... is this still an issue for you?
@misterdjules ... I see there's been recent activity around armv6, what's the status on this one?

**Comment by fivdi** [comment]
This was fixed in io.js. See: https://github.com/nodejs/io.js/issues/283

**Comment by jasnell** [comment]
Ok, given that I'm inclined to defer this to the converged repo and pick up the fix there.

**Comment by fivdi** [comment]
@jasnell Maybe it would be a good idea to defer everything related to ARMv6 and ARMv7 to the converged repo and pick up the fix there. This would include all open issues related to the Raspberry Pi.

**Comment by jasnell** [comment]
@fivdi ... I'm inclined to agree

**Comment by AAADiver** [comment]
Hi - I'm a complete noob so sorry for dumb question. I'm trying to get NPM running on my Pi B+, rasbian, w/Node.JS v12.7. It's failing with Illegal Instructions. I see the patch above (v8) but have no idea how to leverage. How do I move this patch to my Pi and execute it?

root@raspberrypi:~/node# sudo curl -L https://npmjs.org/install.sh | sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   193  100   193    0     0    758      0 --:--:-- --:--:-- --:--:--  1096
100  6239  100  6239    0     0  11501      0 --:--:-- --:--:-- --:--:--  380k
tar=/bin/tar
version:
tar (GNU tar) 1.26
Copyright (C) 2011 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
install npm@latest
fetching: http://registry.npmjs.org/npm/-/npm-2.13.5.tgz
Illegal instruction
Illegal instruction
Illegal instruction
Illegal instruction
It failed

**Comment by fivdi** [comment]
A patched version of v0.12.6 is available at http://node-arm.herokuapp.com. Alternatively io.js can be used.

**Comment by DrCord** [comment]
This issue seems to persist on the 4.x branch of nodejs.

**Comment by bnoordhuis** [comment]
By some strange accident of fate a patch landed today that should take care of the VFPv2/VFPv3 issue: nodejs/node@17665af

**Comment by DrCord** [comment]
How would I go about applying that to Raspbian Jessie?

**Comment by fivdi** [comment]
If the issue is on a Raspberry Pi B+ as reported [at NodeSource](https://github.com/nodesource/distributions/issues/145), then I'd say it's because the Raspberry Pi B+ has an ARMv6 processor, but NodeSource only provide binaries for ARMv7 processors like the Raspberry Pi 2 B. The armv6l binaries at https://nodejs.org/dist/v4.1.2/ should work on the Raspberry Pi B+.

**Comment by DrCord** [comment]
ok, great, thanks.

**Comment by Fishrock123** [comment]
0.x is EOL and if this isn't fixed in 0.12.18 (which is now out) it will never be officially. All 4.x+ releases are tested in CI vs ARMv6.
